mesonbuild/compilers/compilers.py

```diff
@@ -295,6 +311,14 @@ def is_library(fname):
                          's': ['-Os'],
                          }
 
+ccrx_optimization_args = {'0': ['-optimize=0'],
+                          'g': ['-debug'],
```

Comment:
This is probably wrong (just like it is wrong on Arm compilers). This optimization level is "optimize without harming debuggability". With GCC it corresponds to `-Og`. On compilers that do not support it it should be the same as level `0`, so in this case `-optimize=0'.

Reply:
Actually, it's possible to add `-size` on on top of optimization level max via `c_args` or something if the user desires so I'll leave it at that. This aligns with GCC where `-Os` is level 2 + size optimizations.